ex9/lib/main.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:flutter/material.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+void main() => runApp(MaterialApp(home: FirestoreFirstDemo(),));
+
+FirestoreFirstDemoState pageState;
+
+class FirestoreFirstDemo extends StatefulWidget {
+  @override
+  FirestoreFirstDemoState createState() {
+    pageState = FirestoreFirstDemoState();
+    return pageState;
+  }
+}
+
+class FirestoreFirstDemoState extends State<FirestoreFirstDemo> {
+  final firestoreInstance =  Firestore .instance;
+  TextEditingController textEditingController = TextEditingController();
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: Text("Firestore Chat!")),
+      body: SingleChildScrollView(
```

Comment:
You don't need a SingleChildScrollView if you use an expanded ListView. The ListView handles scrolling for you. Can you remove the SingleChildScrollView?